src/widgets/MetricsSelector/hooks/useMetrics.ts

```diff
@@ -35,8 +35,15 @@ export const useMetrics = (id: WidgetId) => {
 
   const presets = useSelector(selectAllPresets);
   const onSelectPreset = useCallback((presetId: PresetId) => {
-    dispatch(setMetricsByWidgetId(id, undefined, presetId, query));
-  }, [query, id, dispatch]);
+    const currentSelectedPreset = presets.find(getPresetIsSelected);
```

Comment:
this kind of logic looks way better in business logic layer (thunks/epics/sagas/whatever)